server/punishments.ts

```diff
@@ -363,6 +369,36 @@ export const Punishments = new class {
 		return FS(SHAREDIPS_FILE).write(buf);
 	}
 
+	/**
+	 * sharedips.tsv is in the format:
+	 * IP, type (in this case always SHARED), note
+	 */
+	async loadSharedIpBlacklist() {
+		const data = await FS(SHAREDIPS_BLACKLIST_FILE).readIfExists();
+		if (!data) return;
+		for (const row of data.split("\n")) {
+			if (!row || row === '\r') continue;
+			const [ip, reason] = row.trim().split("\t");
+			if (!ip.includes('.')) continue;
+			if (!reason) continue;
+
+			Punishments.sharedIpBlacklist.set(ip, reason);
+		}
+	}
+
+	appendSharedIpBlacklist(ip: string, reason: string) {
+		const buf = `${ip}\t${reason}\r\n`;
+		return FS(SHAREDIPS_BLACKLIST_FILE).append(buf);
+	}
+
+	saveSharedIpBlacklist() {
+		let buf = `IP\tReason\r\n`;
+		Punishments.sharedIpBlacklist.forEach((reason, ip) => {
+			buf += `${ip}\t${reason}\r\n`;
```

Comment:
Why are we defaulting to CRLF line endings here? My understanding was that PS defaulted to UNIX line endings (given that that's what main runs on), but should be able to handle CRLF as well.

Reply:
Again, unsure, was copying the other save functions

Reply:
I dont know much about this tbh but I don't think inserting `\r` is good anywhere.

Reply:
I think this would be something to do in another pullreq just because it's not entirely relevant to this

Reply:
Fair enough.